content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+4. The Socket Class builder will try to connect the client to the server and port number provided. The client now has a socket object that, if communication is established, may connect with the server.
+5. The accept() method on the server produces a reference to a new server socket that is connected to the client's socket.
+ 
+If the connection becomes a success, communication can therefore be established using **I/O streams**. Each socket has an input and output stream. The client's OutputStream connects to an InputStream of the server while the server's OutputStream connects to the client's InputStream.
+
+A stream is a collection of data in a particular order. Character streams can be divided into two categories:
+1. Character stream (usually used with text files)
+2. Byte stream(usually used with images)
+Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU.
+
+Our data/messages will be sent across both streams at the same time because TCP is a two-way communication protocol.
+   
+### Programming on Server-side 
+A server socket is an instance of ServerSocket class and can be created by one of the following constructors: 
+
+```java
+serversocket= new Serversocket ( int port )
+```
+
+port: port number at which the server will be listening for requests from clients.
+
+The ServerSocket object sits and waits for network requests to arrive. Our client will try to connect to port 5000 that we created. 
+
+**Note**Make sure the ports are similar otherwise there will be no connection
+Two while loops will be in use: 
+
+**First Loop**- This is to ensure the server is constantly running.
+
+**Second Loop**- Ensures that once the client is connected, the server constantly interacts with the client until the client disconnects.
+
+```java
+while (true) {  
+    try {  
+   socket = serversocket.accept();  
+   
+   inputStreamReader = new InputStreamReader(socket.getInputStream());  
+   outputStreamWriter = new OutputStreamWriter(socket.getOutputStream()); 
+   
+   bufferedReader = new BufferedReader(inputStreamReader);  
+   bufferedWriter = new BufferedWriter(outputStreamWriter);  
+  while (true) { 
+
+  String msgFromClient = bufferedReader.readLine();  
+  
+  System.out.println("Client: " + msgFromClient); 
+
+  bufferedWriter.write(" MSG Received");  
+  bufferedWriter.newLine();  
+  bufferedWriter.flush();  
+  
+ if (msgFromClient.equalsIgnoreCase("BYE"))  
+                break;  
+  
+  }
+      socket.close();  
+      inputStreamReader.close();  
+      outputStreamWriter.close();  
+      bufferedReader.close();  
+      bufferedWriter.close();  
+  
+    } catch (IOException e) {  
+        e.printStackTrace();  
+  }  
+}
+```
+   
+### Methods used in ServerSocket
+
+-` public InputStream( getInputStream())` - returns an InputStream object that receives data.
```

Comment:
I would personally remove the brackets around the method names for consistency and just because it looks like a real method declaration. Example: 

change "public InputStream( getInputStream())" to "public InputStream getInputStream()"